src/Utilities/Compiler/SmallDictionary.cs

```diff
@@ -541,7 +541,10 @@ public Enumerator(SmallDictionary<K, V> dict)
                     }
                 }
 
-                public K Current => _current.Key;
+                public K Current
+                    => _current is null
+                        ? throw new InvalidOperationException()
+                        : _current.Key;
```

Comment:
❗ The previous code is the expected form. The result of the `Current` property is unspecified when does not follow a `MoveNext()` call that returned true.

This applies to all three cases from the PR.

Reply:
If I revert this part I end up with a `CS8602` except if I do revert the nullability of the field in which case it's back to square 1 with `CS8618`.

Reply:
@sharwell What's your preference to fix the error on the property? Shall I just drop this PR?

Reply:
Either of the following is acceptable:

1. Make `_current` non-nullable, which is an accurate representation of its state after `MoveNext()` returns true
2. Return `_current!.Key`